presto-main/src/main/java/com/facebook/presto/metadata/FunctionAndTypeManager.java

```diff
@@ -589,26 +607,33 @@ public boolean nullIfSpecialFormEnabled()
         return !nativeExecution;
     }
 
+    public FunctionHandle lookupFunction(String name, List<TypeSignatureProvider> parameterTypes)
+    {
+        QualifiedObjectName functionName = getFunctionAndTypeResolver().qualifyObjectName(QualifiedName.of(name));
+        return lookupFunction(functionName, parameterTypes);
+    }
+
     /**
      * Lookup up a function with name and fully bound types. This can only be used for builtin functions. {@link #resolveFunction(Optional, Optional, QualifiedObjectName, List)}
      * should be used for dynamically registered functions.
      *
      * @throws PrestoException if function could not be found
      */
-    public FunctionHandle lookupFunction(String name, List<TypeSignatureProvider> parameterTypes)
+    public FunctionHandle lookupFunction(QualifiedObjectName functionName, List<TypeSignatureProvider> parameterTypes)
     {
-        QualifiedObjectName functionName = getFunctionAndTypeResolver().qualifyObjectName(QualifiedName.of(name));
+        Optional<FunctionNamespaceManager<?>> functionNamespaceManager = getServingFunctionNamespaceManager(functionName.getCatalogSchemaName());
+        checkArgument(functionNamespaceManager.isPresent(), "Cannot find function namespace for function '%s'", functionName);
         if (parameterTypes.stream().noneMatch(TypeSignatureProvider::hasDependency)) {
             return lookupCachedFunction(functionName, parameterTypes);
```

Comment:
there's too many assumptions going on here now. this will throw an exception later if the fucntionNamepsaceManager here is not the DEFAULT or currentDefault.